vite.config.ts

```diff
@@ -0,0 +1,36 @@
+import { vitePlugin as remix } from '@remix-run/dev'
+import { flatRoutes } from 'remix-flat-routes'
+import { defineConfig } from 'vite'
+
+const MODE = process.env.NODE_ENV
+
+export default defineConfig({
+	build: {
+		cssMinify: MODE === 'production',
+		rollupOptions: {
+			external: [/node:.*/, 'stream', 'crypto', 'fsevents'],
+		},
+	},
+	plugins: [
+		remix({
+			ignoredRouteFiles: ['**/*'],
+			serverModuleFormat: 'esm',
+			routes: async defineRoutes => {
+				return flatRoutes('routes', defineRoutes, {
+					ignoredRouteFiles: [
+						'.*',
+						'**/*.css',
+						'**/*.test.{js,jsx,ts,tsx}',
+						'**/__*.*',
+						// This is for server-side utilities you want to colocate next to
+						// your routes without making an additional directory.
+						// If you need a route that includes "server" or "client" in the
+						// filename, use the escape brackets like: my-route.[server].tsx
+						'**/*.server.*',
+						'**/*.client.*',
```

Comment:
This is probably the most significant *change*. It's the primary reason source code was changed. Remix with vite is more strict about your code that can end up in the client not using anything that's server-only so if you export anything other than the standard exports in a route that uses server-only code it'll error.